test/functional/test_runner.py

```diff
@@ -611,6 +617,34 @@ def get_next(self):
                         status = "Passed"
                     elif proc.returncode == TEST_EXIT_SKIPPED:
                         status = "Skipped"
+                    elif attempt < self.attempts:
+                        # cleanup
+                        if self.use_term_control:
+                            clearline = '\r' + (' ' * dot_count) + '\r'
+                            print(clearline, end='', flush=True)
+                        dot_count = 0
+                        shutil.rmtree(testdir, ignore_errors=True)
+                        self.jobs.remove(job)
+                        print("{} failed at attempt {}/{}, Duration: {} s".format(name, attempt, self.attempts, int(time.time() - start_time)))
```

Comment:
nit: I think in modern python better to use f-strings literals:
```
print(f"{name} failed at attemp {attempt}/{self.attempts}, Duration: {int(time.time() - start_time} s"
```

`.format` is generally more powerful but f-strings literals make code more easy to read IMO

Reply:
interesting. but we use `.format` throughout the code so I'd rather stick to it for now :)

Reply:
(I'd much prefer f-strings 🙈 )

Reply:
2:1 ok :D